nifi-nar-bundles/nifi-iceberg-bundle/nifi-iceberg-processors/src/main/java/org/apache/nifi/processors/iceberg/converter/GenericDataConverters.java

```diff
@@ -245,9 +250,14 @@ static class RecordConverter extends DataConverter<Record, GenericRecord> {
 
             for (DataConverter<?, ?> converter : converters) {
                 final Optional<RecordField> recordField = recordSchema.getField(converter.getSourceFieldName());
-                final RecordField field = recordField.get();
-                // creates a record field accessor for every data converter
-                getters.put(converter.getTargetFieldName(), createFieldGetter(field.getDataType(), field.getFieldName(), field.isNullable()));
+                if (recordField.isEmpty()) {
+                    Types.NestedField missingField = schema.field(converter.getSourceFieldName());
```

Comment:
Here you should get the `targetFieldName` since it contains the iceberg schema field's name.
```suggestion
                    Types.NestedField missingField = schema.field(converter.getTargetFieldName());
```